### PR TITLE
Support Multiple Newsletters

### LINKED ISSUE #30: Multiple Newsletter
Hi!
Is it possible to finally have multiple newsletters? Or is it going to be difficult?
Thanks for your work.

### PR DESCRIPTION
**Pull request by ItzNotABug**
This PR adds the ability to manage multiple newsletters directly from Ghosler.

---
## The Flow
1. User publishes a blog post
2. Ghosler receives the data via webhook
3. Ghosler checks if there are more than 1 newsletters -
  a. If no, sends the newsletter as usual to all subscribed members.
  b. If yes, saves the Post data with all the required information for further actions.
4. User must go to Analytics page
5. Select the new post which will now have a `Send` button
6. A list of all `active` newsletters is shown
7. Click `Send` on a newsletter & the Post will be sent to the subscribers of that newsletter only

### CODE REVIEW

File: routes/newsletters.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+router.get('/:postId', async (req, res) => {
+    const postId = req.params.postId;
+
+    if (!postId) {
+        return res.render('dashboard/newsletters', {
+            level: 'error',
+            message: 'Post Id missing!'
+        });
+    }
```
Comment (ItzNotABug) [inline routes/newsletters.js]: I don't think this is necessary here afaik, we already have a redirect on `/newsletters/` endpoint.

File: utils/api/ghost.js [hunk 2]
```diff
@@ -19,6 +19,30 @@ export default class Ghost {
...
     }
 
+    /**
+     * Returns the number of 'active' newsletters.
+     *
+     * @returns {Promise<number>} The number of active newsletters.
+     */
+    async newslettersCount() {
+        const newsletters = await this.newsletters();
+        return newsletters.meta?.pagination?.total ?? 0;
```
Comment (ItzNotABug) [inline utils/api/ghost.js:42]: Default value should be `1` due to this - https://github.com/ItzNotABug/ghosler/blob/a507c216b53cfe14e2cb1d8b36b47d6b3fd3dfc0/routes/published.js#L35

Comment (ItzNotABug) [reply utils/api/ghost.js:42]: On a second thought, maybe we should use `0` & simply not send emails if there are no active newsletters. One newsletter must exist.

**Comment by ItzNotABug** [comment]
Adding the list of 'active' newsletter along with a Generic newsletter type that sends the email to all members.

ghosler-newsletters-selection

**Comment by ItzNotABug** [comment]
I have tested this with the latest commit and this seems to work fine. This can be merged.

**Comment by fritz-fritz** [comment]
Would be an interesting idea to setup tags for the newsletters ( eg `#newsletter1`, `#weeklydigest`, etc... ) automatically over the api and that way when posts are published and have been tagged with the appropriate internal tag, it gets mailed to the corresponding newsletter automatically! I'm not currently using multiple newsletters so am not inclined to implement this at the moment but thought I'd share the idea!

**Pull request merged**